### PR TITLE
Origin/py 3 8 warns rebase changes

### PR DESCRIPTION
**Pull request by franchuterivera**
Deprecated utilities:
sklearn.utils.testing is deprecated --> numpy assert equivalent
sklearn.externals.joblib -->use from joblib directly
sklearn.linear_model.stochastic_gradient --> from sklearn.linear_model directly
Setting a random_state has no effect since shuffle is False... --> only add random state is no shuffle
DeprecationWarning: Please use assertRaisesRegex instead. --> Removed the "p" as it is a deprecated alias https://docs.python.org/3.2/library/unittest.html
"DeprecationWarning: invalid escape sequence ( ---> re.compile('(') --> Added another backslash re.compile('\(')

### CODE REVIEW

**Comment by franchuterivera** [comment]
I see failures in Travis on kernel pca, but not on a manual run:
(venv_autosklearn3_8) chico@killua:~/work/auto-sklearn_fork$ python -m pytest -v test/test_pipeline/components/feature_preprocessing/test_kernel_pca.py
...
**============================= 3 passed, 1 skipped, 2 warnings in 2.55s ==============================**

(venv_autosklearn3_8) chico@killua:~/work/auto-sklearn_fork$ git log --name-status HEAD^..HEAD
commit 39020af3dd8b194f89e56defa244322c69118160 (HEAD -> origin/py_3_8_warns_rebase_changes, origin/origin/py_3_8_warns_rebase_changes)
Author: chico <francisco.rivera.valverde@gmail.com>
Date:   Fri Apr 17 18:08:05 2020 +0200

    Fixing flake erors

M       test/test_automl/test_estimators.py
M       test/test_pipeline/implementations/test_SparseOneHotEncoder.py

**Comment by mfeurer** [comment]
That KernelPCA error is a bug in scikit-learn that will be fixed in the next release, see https://github.com/automl/auto-sklearn/issues/734
